Dockerfile.commonv8_3

```diff
@@ -217,23 +190,56 @@ RUN apt-get update && \
     rm -rf build \
     opencv-4.5.1 \
     opencv_contrib-4.5.1
+    
+    
+# 
+# upgrade cmake - https://stackoverflow.com/a/56690743
+# this is needed to build some of the ROS2 packages
+#
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+          libssl-dev \
+		  software-properties-common \
+		  apt-transport-https \
+		  ca-certificates \
+		  gnupg \
+    && rm -rf /var/lib/apt/lists/* \
+    && apt-get clean
+		  	  
+# install cmake
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.23.0-rc2/cmake-3.23.0-rc2.tar.gz && \
+    tar -xzf cmake-3.23.0-rc2.tar.gz && \
+    cd cmake-3.23.0-rc2 && cmake . -DCMAKE_BUILD_TYPE=Release && \
+    make -j 12 && \
+    make install && \
+    rm -rf cmake-3.23.0-rc2.tar.gz && \
+    rm -rf cmake-3.23.0-rc2
 
 ###yaml-cpp
 ## compile yaml-cpp-0.6, which some ROS packages may use (but is not in the 18.04 apt repo)
 RUN git clone --branch yaml-cpp-0.6.0 https://github.com/jbeder/yaml-cpp yaml-cpp-0.6 && \
     cd yaml-cpp-0.6 && \
     mkdir build && \
     cd build && \
-    cmake -DBUILD_SHARED_LIBS=ON .. && \
+    cmake -DBUILD_SHARED_LIBS=ON -DYAML_CPP_INSTALL=ON .. && \
     make -j$(nproc) && \
-    cp libyaml-cpp.so.0.6.0 /usr/lib/aarch64-linux-gnu/ && \
-    ln -s /usr/lib/aarch64-linux-gnu/libyaml-cpp.so.0.6.0 /usr/lib/aarch64-linux-gnu/libyaml-cpp.so.0.6
+    make install
+    #cp libyaml-cpp.so.0.6.0 /usr/lib/aarch64-linux-gnu/ && \
+    #ln -s /usr/lib/aarch64-linux-gnu/libyaml-cpp.so.0.6.0 /usr/lib/aarch64-linux-gnu/libyaml-cpp.so.0.6
 
-### ROS
```

Comment:
This sounds like it was fun to figure out

Reply:
The old "steal" from dusty-nv's jetson containers